frontend/src/metabase/app.js

```diff
@@ -53,7 +54,10 @@ function init() {
     // we shouldn't redirect these URLs because we want to handle them differently
     let WHITELIST_FORBIDDEN_URLS = [
         /api\/card\/\d+\/query$/,
-        /api\/database\/\d+\/metadata$/
+        /api\/database\/\d+\/metadata$/,
+        /api\/database\/\d+\/fields/,
+        /api\/table\/\d+\/query_metadata$/,
+        /api\/table\/\d+\/fks$/
```

Comment:
this seems to conflate an un-authenticated user with a user who is not authorized for the specific url they are trying to access. should we be using 401 for un-authenticated users and reserve 403 for resources that are forbidden for the specific user (or anonymous user)?